spec/lograge_spec.rb

```diff
@@ -21,24 +23,34 @@
         Lograge.remove_existing_log_subscriptions
       end.to change {
         Lograge.notification_listeners_for('process_action.action_controller')
-      }
+      }.to([])
     end
 
     it 'removes subscribers for all events' do
       expect do
         Lograge.remove_existing_log_subscriptions
       end.to change {
         Lograge.notification_listeners_for('render_template.action_view')
-      }
+      }.to([])
     end
 
-    it "does not remove subscribers that aren't from Rails" do
-      blk = -> {}
-      ActiveSupport::Notifications.subscribe('process_action.action_controller', &blk)
-      Lograge.remove_existing_log_subscriptions
-      listeners = Lograge.notification_listeners_for('process_action.action_controller')
-      expect(listeners.size).to eq(1)
+    shared_examples 'preserving non-Rails subscribers' do |event_name|
+      context "with event_name #{event_name}" do
+        it "does not remove subscribers that aren't from Rails" do
+          proc_subscriber = ActiveSupport::Notifications.subscribe(event_name, proc {})
+          custom_subscriber =
+            ActiveSupport::Notifications.subscribe(event_name, CustomListener.new)
```

Comment:
Also check a custom class, in case there was some special handling of `Proc` instances